libsolidity/analysis/TypeChecker.cpp

```diff
@@ -366,8 +366,16 @@ bool TypeChecker::visit(FunctionDefinition const& _function)
 		else
 		{
 			solAssert(type(var)->nameable(), "");
-			if (type(var)->containsNestedMapping() && _function.isPublic())
-				m_errorReporter.typeError(3312_error, var.location(), "Type is required to live outside storage.");
+			if (
```

Comment:
I think you can replace the check for `== Type::Category::Mapping` above by `containsNestedMapping`. Since it is a variable declaration, the type has to be nameable.